test/libyul/StackReuseCodegen.cpp

```diff
@@ -345,6 +345,20 @@ BOOST_AUTO_TEST_CASE(reuse_slots_function_with_gaps)
 	);
 }
 
+BOOST_AUTO_TEST_CASE(reuse_on_decl_assign_to_last_used)
+{
+	string in = R"({
+		let x := 5
+		let y := x // y should reuse the stack slot of x and this should not generate any code
```

Comment:
What about things like `add(x, 2)`?

Reply:
Got that working as well now, following your suggestion - added a test for it!

Reply:
The comment is wrong now.